spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/config/DataFlowControllerAutoConfiguration.java

```diff
@@ -300,8 +300,8 @@ public TaskExecutionController taskExecutionController(
 		}
 
 		@Bean
-		public TaskExecutionThinController taskExecutionThinController(AggregateTaskExplorer aggregateTaskExplorer, TaskJobService taskJobService) {
-			return new TaskExecutionThinController(aggregateTaskExplorer, taskJobService);
+		public TaskExecutionThinController taskExecutionThinController(AggregateTaskExplorer aggregateTaskExplorer, TaskDefinitionRepository taskDefinitionRepository, TaskJobService taskJobService) {
```

Comment:
What is this change for?

Reply:
nm I see below

Reply:
If the search is for a task where we have no definition of that name then return 404. Otherwise the result is an empty list.